### PR TITLE
Implement bayesian revenue math

### PR DESCRIPTION
**Pull request by valentino-amadeus**
#### What is the purpose of this pull request?
Implement mathematical analysis for the bayesian revenue test.

#### How should this be manually tested?
Call the functions from `decisionRule/bayesianRevenue.ts` with a bunch of random (positive integer) parameters and compare the results with what they should be - the functions from `decisionRule/bayesianConversion.ts` may be useful in the process.

#### Types of changes
- [x] Math stuff (non-breaking change which also doesn't add any functionality (yet))
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: node/utils/mathTools/digamma-function.ts [hunk 1]
```diff
@@ -1,125 +0,0 @@
...
-        tmp += 1 / x
-    }
-    // If x < 1, use recurrence to shift to > 1..
-    while (x < 1) {
-        tmp -= 1 / x
-        x += 1
-    }
-    tmp += digammaRationalApprox(x)
-    return tmp
-}
```
Comment (tergol1) [inline node/utils/mathTools/digamma-function.ts:125]: I think we already discussed this. So just let these code here and we think about them in the future (my guess is that we'll use'em)

File: node/utils/mathTools/polynomial-evaluation.ts [hunk 2]
```diff
@@ -1,26 +0,0 @@
...
-    }
-    return num
-}
-
-export function evalrational(P: number[], Q: number[], x: number): number {
-    const num = evalpoly(P, x)
-    const den = evalpoly(Q, x)
-
-    return num / den
-}
```
Comment (tergol1) [inline node/utils/mathTools/polynomial-evaluation.ts:26]: Don't remove this as well

File: node/utils/mathTools/statistics/betaDistribution.ts [hunk 3]
```diff
@@ -11,8 +11,4 @@ export function incompleteBeta(x: number, a: number, b: number) {
     }
 
     return value * beta(a, b)
-}
-
-export function intermediateBeta(x: number, y: number, a: number, b: number) {
-    return Math.abs(incompleteBeta(y, a, b) - incompleteBeta(x, a, b))
-}
```
Comment (tergol1) [inline node/utils/mathTools/statistics/betaDistribution.ts:18]: Don't remove this as well

File: node/utils/randomExploration.ts [hunk 4]
```diff
@@ -1,4 +1,4 @@
-import { RandomBeta } from './mathTools/statistics/betaSampling'
+import { RandomBeta } from './mathTools/forBetaDistribution/statistics/betaSampling'
```
Comment (tergol1) [inline node/utils/randomExploration.ts:1]: One example of bad naming I used in the past haha
We can try to change this in the future

File: node/utils/mathTools/decisionRule/bayesianRevenue.ts [hunk 5]
```diff
@@ -0,0 +1,49 @@
+import { probabilityUnderDiagonal, differenceExpectationUnderDiagonal } from '../riemannIntegration'
```
Comment (tergol1) [inline node/utils/mathTools/decisionRule/bayesianRevenue.ts:1]: Commenting here but it's just to say that it's a great satisfaction to review this commit :satisfaction:

**Review by tergol1 — APPROVED** [review]
LGTM!
Just be careful to undo the deleting of files we already agreed to don't remove.

File: node/utils/randomExploration.ts (hunk 4 above)
Comment (valentino-amadeus) [reply node/utils/randomExploration.ts:1]: Absolutely! https://github.com/vtex/ab-tester/pull/84#discussion_r605940258

File: node/utils/mathTools/digamma-function.ts (hunk 1 above)
Comment (valentino-amadeus) [reply node/utils/mathTools/digamma-function.ts:125]: We discussed this way after this PR was opened hehe.

File: node/utils/mathTools/statistics/betaDistribution.ts (hunk 3 above)
Comment (valentino-amadeus) [reply node/utils/mathTools/statistics/betaDistribution.ts:18]: Ok :okay:

**Comment by valentino-amadeus** [comment]
Done. [#133](https://github.com/vtex/ab-tester/pull/133)

**Pull request merged**